src/helpers/url.tsx

```
@@ -140,3 +140,111 @@ export function removeQueryParamAndRedirect(
+		default:
+			return cleanedUrl; // Return cleaned URL for generic websites or unsupported social media
+	}
+};
+
+// Function to extract username from URL based on the regex pattern
+export const extractUsernameFromPattern = (
+	url: string,
+	regex: RegExp,
+): string => {
+	const match = url.match(regex);
+	if (match && match[1]) {
```

Biome

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)